### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by CarlosE-Feliz**
Your console app should behave in the same way as in the example. ✔
Create a main.rb file that will serve as your console app entry-point. ✔
In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. ✔

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    name = gets.chomp
+
+    person =
+      case person_type
+      when '1'
+        print 'Has parent permission? [Y/N]: '
+        # *If needed, ask for parameters for the option.
+        parent_permission = gets.chomp
+        parent_permission = parent_permission.downcase == 'y'
```
Comment (hiromataba) [inline main.rb]: - [x] I am not able to create a student successfully, Whenever I enter the `parent_permission` I am getting and error, Kindly fix that and each person you are creating should have an `ID`

image

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+        Student.new(age, name, parent_permission)
+      when '2'
+        print 'Specialization: '
+
+        specialization = gets.chomp
+
+        Teacher.new(age, specialization, name)
+      end
```
Comment (hiromataba) [inline main.rb]: - [x] I am not able to create a Teacher successfully, Whenever I enter the `Specialization` I am getting an error, Kindly fix that and each person you are creating should have an `ID`

image

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,151 @@
+require './classroom'
+require './person'
+require './rental'
+require './student'
+require './teacher'
+require './book'
+# rubocop:disable all
```
Comment (hiromataba) [inline main.rb:7]: - [x] Linters are used to make the app professional and help you write clean code, so kindly avoid disabling the linters

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    print 'Title: '
+
+    title = gets.chomp
+
+    print 'Author: '
+
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created'
```
Comment (hiromataba) [inline main.rb]: - [x] [OPTIONAL] You may add successfully after book created and the whole message should be `Book Created  Successfully`

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+
+    book_index = gets.chomp.to_i
+
+    puts "\nSelect a person from the following list by number (not id)"
```
Comment (hiromataba) [inline main.rb:77]: - [x] I am not able to create a rental whenever I am trying to create one I am getting this error, Please fix that

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi :wave: @CarlosE-Feliz , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### TO highlight :champagne: 
- No linter errors :heavy_check_mark: 
- Nice PR description and title :+1: 

## Required Changes ♻️

- [ ] Please add commands and instructions to run the application in the terminal in your README file

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (CarlosE-Feliz) [reply main.rb]: Thanks for your feedback

File: main.rb (hunk 3 above)
Comment (CarlosE-Feliz) [reply main.rb:7]: Thanks for your feedback

File: main.rb (hunk 4 above)
Comment (CarlosE-Feliz) [reply main.rb]: Thanks for your feedback

File: main.rb (hunk 5 above)
Comment (CarlosE-Feliz) [reply main.rb:77]: Thanks for your feedback

File: main.rb (hunk 2 above)
Comment (CarlosE-Feliz) [reply main.rb]: Thanks for your feedback!

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi @CarlosE-Feliz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**